### PR TITLE
Freature/pagination

### PR DESCRIPTION
**Pull request by schittlur**

### CODE REVIEW

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 1]
```diff
@@ -13,22 +14,31 @@ const getParsedQuery = (location) => {
...
 
     const [searchInputState, setSearchInputState] = useState({
         input: ''
     });
-
+    
     // const jsonObj = dummyJSON;
     let location = useLocation();
-    const api_url = 'http://localhost:8080/es_api/?keyword='
+    const api_url = 'http://localhost:8080/es-api/?keyword='
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:38]: This endpoint specification should be moved into configuration

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 2]
```diff
@@ -13,22 +14,31 @@ const getParsedQuery = (location) => {
...
     const parsedQuery = getParsedQuery(location);
+    const pageNum = getPage(location);
+    console.log(pageNum);
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:41]: Need to remove debugging statements before merging

File: src/components/SearchResultsPage/SearchResultsPage.js [hunk 3]
```diff
@@ -150,6 +162,16 @@ const SearchResultPage = (props) => {
...
         )
     }
+    
+    let pagination =  null;
+    if (coursesState.coursesObj && !coursesState.isLoading) {
+        pagination = (
+            <Pagination 
+            courseState={coursesState.coursesObj} page={coursesState.page} searchInputState={searchInputState.input}
+             />
+            // totalPages={totalPages} onLeftClick={onLeftClick} onRightClick={onRightClick} onPageClick={handlePageClick}
```
Comment (mattdrayer) [inline src/components/SearchResultsPage/SearchResultsPage.js:172]: Need to remove commented-out code before merging

**Review by mattdrayer — APPROVED** [review]
Just a couple small tweaks -- looks good!

File: src/components/SearchResultsPage/SearchResultsPage.js (hunk 1 above)
Comment (schittlur) [reply src/components/SearchResultsPage/SearchResultsPage.js:38]: Mac will handle this in his commits

File: package.json [hunk 4]
```diff
@@ -8,9 +8,11 @@
     "@testing-library/user-event": "^12.1.10",
     "axios": "^0.21.1",
     "query-string": "^6.14.0",
+    "rc-pagination": "^3.1.6",
```
Comment (jdiouf) [inline package.json:11]: I think we're not using these dependencies anymore? We can remove them

File: package.json [hunk 5]
```diff
@@ -8,9 +8,11 @@
...
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "react-loader-spinner": "^4.0.0",
+    "react-paginate": "^7.1.0",
```
Comment (jdiouf) [inline package.json:15]: We can remove this as well if not in use

File: src/components/Pagination/Pagination.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+import React from 'react'
+import classes from './Pagination.module.css';
+import { Link } from 'react-router-dom'
+
+
```
Comment (jdiouf) [inline src/components/Pagination/Pagination.js:5]: Let's add a comment here describing what this component does, and some of the logic

**Review by jdiouf — CHANGES_REQUESTED** [review]

**Review by jdiouf — APPROVED** [review]

**Pull request merged**